tests/apps/namd/namd_test.py

```diff
@@ -0,0 +1,172 @@
+        self.valid_prog_environs = ['builtin']
+        super().__init__('cpu')
+
+        self.scale_reference = {
+            '1': {
+                'leibniz:single-node': {'days_ns': (0.345, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.1991375, None, 0.05, 'days/ns')},
+            },
+            '2': {
+                'leibniz:single-node': {'days_ns': (0.24671949, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.1601965, None, 0.05, 'days/ns')},
+            },
+            '4': {
+                'leibniz:single-node': {'days_ns': (1.317, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.7218169, None, 0.05, 'days/ns')},
+            },
+        }
+        self.reference = self.scale_reference[self.num_nodes]
+
+    @run_after('setup')
+    def set_num_cpus(self):
+        self.num_tasks = int(self.num_nodes)
+
+        configFile = self.download_material()
+
+        # VSC specific config
+        if self.current_system.name == 'leibniz':
+            self.num_cpus_per_task = 28
+            self.modules = ['NAMD/2.14-verbs-smp']
+        elif self.current_system.name == 'vaughan':
+            self.num_cpus_per_task = 64
+            self.modules = ['NAMD/2.14-verbs-smp']
+        elif self.current_system.name == 'genius':
+            self.num_cpus_per_task = 36
+            self.modules = ['NAMD/2.14-foss-2019b-mpi']
+
+        self.create_nodelist()
+
+        self.executable = f'charmrun +p {self.num_cpus_per_task*self.num_tasks} $EBROOTNAMD/namd2 {configFile}/{configFile}.namd'
+
+
+@rfm.simple_test
+class Namd_NotSMP_CPUTest(NamdBaseTest):
+    # NAMD notSMP CPU test
+
+    def __init__(self):
+        self.time_limit = '20m'
+
+        self.valid_systems = ['leibniz:single-node',
+                              'vaughan:single-node',
+                              'genius:single-node',
+                              'hydra:single-node']
+
+        self.valid_prog_environs = ['builtin']
+        super().__init__('cpu')
+
+        self.scale_reference = {
+            '1': {
+                'leibniz:single-node': {'days_ns': (0.347779, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.188093, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.202701, None, 0.05, 'days/ns')},
+            },
+            '2': {
+                'leibniz:single-node': {'days_ns': (0.1782715, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.09856985, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.1099565, None, 0.05, 'days/ns')},
+            },
+            '4': {
+                'leibniz:single-node': {'days_ns': (1.05726, None, 0.05, 'days/ns')},
+                'vaughan:single-node': {'days_ns': (0.5438339, None, 0.05, 'days/ns')},
+                'hydra:single-node': {'days_ns': (0.5427335, None, 0.05, 'days/ns')},
```

Comment:
it seems that running in 4 nodes tanks efficiency, maybe we should drop it?

Reply:
On 4 nodes it is actually a different test, uses a different simulation, a bigger one.

Reply:
ah ok, no problem then